src/wmtk/TriMeshOperationExecutor.hpp

```diff
@@ -52,56 +52,77 @@ class TriMesh::TriMeshOperationExecutor
+     */
+    static const std::array<std::vector<long>, 3> get_split_simplices_to_delete(
+        const Tuple& tuple,
+        const TriMesh& m);
 
-    const std::array<long, 2>& incident_vids() const { return m_incident_vids; }
+    /**
+     * @brief gather all simplices that are deleted in a collapse
+     *
+     * The deleted simplices are the intersection of the open star of the vertex and the closed star
+     * of the edge. This comes down to one vertex, three edges, and two faces if the edge is on the
+     * interior. On the boundary it is one vertex, two edges, and one face.
+     */
+    static const std::array<std::vector<long>, 3> get_collapse_simplices_to_delete(
+        const Tuple& tuple,
+        const TriMesh& m);
+
+    std::vector<IncidentFaceData>& incident_face_datas() { return m_incident_face_datas; }
+
+    const std::array<long, 2>& incident_vids() const { return m_spine_vids; }
 
     const long operating_edge_id() const { return m_operating_edge_id; }
 
-    void glue_ear_to_face(
+    void update_ids_in_ear(
         const long ear_fid,
-        const long new_face_fid,
+        const long new_fid,
         const long old_fid,
-        const long eid);
+        const long new_eid);
 
-    void merge(const long& new_vid);
+    void connect_ears();
     Tuple split_edge();
 
+    Tuple collapse_edge();
+
     /**
      * @brief
      *
      * @param new_fids_top the two new fids on the top sides of AB generated by split_edge(AB)
      * @param new_fids_bottom the two new fids on the bottom sides of AB generated by split_edge(AB)
      */
     // return the two new fids in order
-    std::array<long, 2> glue_new_triangle_topology(
+    void replace_incident_face(
         const long new_vid,
-        const std::vector<long>& replacement_eids,
-        const IncidentFaceData& face_data);
-    void glue_new_faces_across_AB(
-        const std::array<long, 2> new_fids_top,
-        const std::array<long, 2> new_fids_bottom);
+        const std::vector<long>& spine_eids,
+        IncidentFaceData& face_data);
+    void connect_faces_across_spine();
     std::vector<long> request_simplex_indices(const PrimitiveType type, long count);
 
-    std::vector<std::vector<long>> simplices_to_delete; // size 3 for vertex, edge, face
-    std::vector<long> cells_to_update_hash;
+    std::array<std::vector<long>, 3> simplex_ids_to_delete;
+    std::vector<long> cell_ids_to_update_hash;
     TriMesh& m_mesh;
     Tuple m_operating_tuple;
 
+    long hash_at_cell(const long cell) { return hash_accessor.scalar_attribute(cell); }
+
 private:
     // common simplicies
-    std::array<long, 2> m_incident_vids; // V_A_id, V_B_id;
+    std::array<long, 2> m_spine_vids; // V_A_id, V_B_id;
     long m_operating_edge_id;
 
     // simplices required per-face
     std::vector<IncidentFaceData> m_incident_face_datas;
 
-    IncidentFaceData get_incident_face_data(const Tuple& t);
+    IncidentFaceData get_incident_face_data(Tuple t);
```

Comment:
I am changing the tuple within the function to align it with the edge. Because I am using a SimplicialComplex to gather the incident faces, there is no guarantee that the tuple is on the edge.